### PR TITLE
Quickscan

### PR DESCRIPTION
**Pull request by jreadey**
This change addresses issue #255 - verbose info not correct.

For the verbose param to return the current info for domain or datasets, you'll need to do a PUT / with rescan and flush params set.  See the checkVerbose function in value_test.py to see how this works.

### CODE REVIEW

File: hsds/datanode.py [hunk 1]
```diff
@@ -165,8 +167,16 @@ async def bucketScan(app):
                 tb = traceback.format_exc()
                 print("traceback:", tb)
 
-        log.info(f"bucketScan - sleep: {async_sleep_time}")
-        await asyncio.sleep(async_sleep_time)
+            last_action = time.time()
+
+        now = time.time()
+        if (now - last_action) > async_sleep_time:
```
Comment (mattjala) [inline hsds/datanode.py:173]: If `last_action` is set to the current time before the scanning loop is performed/at the end of each scan loop, won't this always be true?

File: hsds/datanode.py [hunk 2]
```diff
@@ -165,8 +167,16 @@ async def bucketScan(app):
...
+            sleep_time = async_sleep_time  # long nap
+        else:
+            sleep_time = short_sleep_time  # shot nap
```
Comment (mattjala) [inline hsds/datanode.py:176]: What's the idea behind varying the time between scans based on how long the scan itself takes? Or is `last_action` tracking something else?

File: admin/config/config.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ log_prefix: null # Prefix text to append to log entries
 max_tcp_connections: 100 # max number of inflight tcp connections
 head_sleep_time: 10 # max sleep time between health checks for head node
 node_sleep_time: 10 # max sleep time between health checks for SN/DN nodes
-async_sleep_time: 10 # max sleep time between async task runs
```
Comment (mattjala) [inline admin/config/config.yml:35]: Why the change in automatic scan frequency? I thought the idea was to allow certain requests to force a scan.

**Review by mattjala — APPROVED** [review]
This enabled `H5Dget_storage_size` to work from the REST VOL, when it requests a flush and rescan first. I don't fully understand the reasoning behind all the scan changes, but we can iron those out later. Approved.

File: hsds/datanode.py (hunk 1 above)
Comment (jreadey) [reply hsds/datanode.py:173]: If there are no root_ids, last_action doesn't get updated, so eventually the if case will get executed.

File: hsds/datanode.py (hunk 2 above)
Comment (jreadey) [reply hsds/datanode.py:176]: If there's nothing going on, use a longer sleep and save cpu cycles.  It's not how long the scan takes but rather the last time there was a domain to scan.

Comment (jreadey) [reply hsds/datanode.py:176]: Here's a quick check using the value_test.py test case.
I modified the testPut1DDataset to do a sequence of flushes and rescans:

        # check values we should get from a verbose query
        import time
        ts = time.time()
        for i in range(10):
            expected = {"num_chunks": 1, "allocated_size": 40}
            self.checkVerbose(dset_id, headers=headers, expected=expected)
            now = time.time()
            elapsed = now - ts
            print(f"flush {elapsed:.2f}")
            ts = now

Output is:

          $ python value_test.py ValueTest.testPut1DDataset
          testPut1DDataset /home/test_user1/hsds_test/valuetest/20231010T154859_160988Z
          flush 2.54
          flush 0.12
          flush 0.12
          flush 0.13
          flush 0.13
          flush 0.13
          flush 0.12
          flush 0.12
          flush 0.13
          flush 0.13


The first round takes a couple of seconds because apparently  the bucketScan task is in long sleep mode.  After the first flush, it "wakes up" and each update only takes about 0.1s.

File: admin/config/config.yml (hunk 3 above)
Comment (jreadey) [reply admin/config/config.yml:35]: The buckeScan task is still running independently, so the sleep is there to not unnecessarily consume cpu.
It might be better to use something like asyncio.Event (see https://docs.python.org/3/library/asyncio-sync.html#asyncio.Event).

Comment (jreadey) [reply admin/config/config.yml:35]: I'll leave using signals  as a future enhancement since what we have should be good enough for this non-critical path.
